pkg/privilege/privileges/cache.go

```diff
@@ -925,11 +925,7 @@ func (p *immutable) decodeUserTableRow(row chunk.Row, fs []*resolve.ResultField)
 				value.AccountLocked = true
 			}
 		case f.ColumnAsName.L == "plugin":
-			if row.GetString(i) != "" {
-				value.AuthPlugin = row.GetString(i)
-			} else {
-				value.AuthPlugin = mysql.AuthNativePassword
-			}
+			value.AuthPlugin = row.GetString(i)
```

Comment:
When `ow.GetString(i) != ""` happen, can we avoid it after this PR?

Reply:
To avoid unexpected cases, I add the barrier back https://github.com/pingcap/tidb/pull/57391/commits/7747b9d868b6d1c48d2634f66c54bdaaaa884174